### PR TITLE
Users able to add a product to the Customer's Cart

### LINKED ISSUE #5: User should be able to add product to customer order
**Given** a customer has been made active in the program
**When** the user selects the option to add a product to the active customer's order
**Then** the user should be prompted with a numbered list of all products
**And** when one is chosen, the product should be added to an order for the active customer

> **Pro tip**: To make it easier to add multiple products, when the user selects a product to add to the customer's order, display the menu of products again. Make sure the last option provides the option to go back to main menu.

```bash
1. Diapers
2. Case of Cracking Cola
3. Bicycle
4. AA Batteries
...
9. Done adding products
```

### PR DESCRIPTION
**Pull request by EliOJackson**
# Description

Users are able to add a product(s) to the Active Customer's Cart using the prompt, after the Active Customer has been selected.

Fixes #5 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Functionality
To begin run the following command
```
node db/build_table.js
npm start
```

**Active Customer Must Be Selected**
Users will not be able to add a Product to the Customers Cart if an active customer has not been selected. If you select #5 from the main menu before choosing an active customer, you will see the following error message: 

```
You cannot add products to the cart until you select an active customer. Please choose an active customer to continue.
```
**Select the product to be added to the cart**
After an active customer has been selected, proceed to the product list by selecting option # 5 on the Welcome Menu. Once selected, the user will be presented with a list of ALL products. The user must then input a number that corresponds to the product they wish to add. If they do not select a number that corresponds to a product, they will receive the following error message: 

```
You did not enter a valid option. Please try again!
```

***Select the quantity of the Product to be added to the Order***
Once a product has been selected, the User will be presented with a prompt to enter a total Quantity. The user will enter a number for the quantity they wish to add to their cart. If the quantity entered is not available, the User will be presented with a message notifying them how many of the product is available. Ex:

```
I'm sorry but there are only X items available to be added to your cart, please enter another quantity.
```
Once the user has entered an available quantity, the will receive a message notifying them of such. If no active order exists for the customer, an order will be created with the product added, and the customer will receive the following message: 

```
Your Order has been created!
```
If an active order already exists for the customer, the product will be added to the order, and the customer will receive the following message: 

```
Your Order has been updated with your product(s)
```
Once a product has been added to an order, the User will be returned to the Welcome Menu. 

# How Has This Been Tested?

In order to run tests:

```
npm test
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/order_product.test.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    after(done => {
+        createOrderProductTable().then(() => {
+            done();
+        });
+    });
+    let newOP = {
+        quantity: 10,
+        order_id: 5,
+        product_id: 6,
+        price: 5
```
Comment (sarahmorris926) [inline test/order_product.test.js]: I think you might need to take out `price` here, and change quantity to `order_quantity`.

File: test/order_product.test.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            console.log('Post Post Order Product Should Return the Line ID of the object posted', err);
+        });
+    });
+    it("should return the object posted when called by the Line ID", () => {
+        let expected = {
+            line_id: 141,
+            order_id: 5,
+            product_id: 6,
+            price: 5,
+            quantity: 10
```
Comment (sarahmorrisokeefe) [inline test/order_product.test.js]: I think you might also need to change quantity here (to say `order_quantity`) and take out `price`.

File: test/order_product.test.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+//POST New Order Product
+describe("post one order proudct", () => {
+    after(done => {
+        createOrderProductTable().then(() => {
+            done();
+        });
+    });
+    let newOP = {
+        quantity: 10,
```
Comment (sarahmorrisokeefe) [inline test/order_product.test.js]: I think you might need to take out `price` here, and change quantity to `order_quantity`.

**Review by sarahmorrisokeefe — CHANGES_REQUESTED** [review]
I left a few line comments in the `order_product.test.js` file. Looks good - just those changes and then all the tests should pass for that file!

**Review by spenceryork — APPROVED** [review]
functionality works and all tests pass when test files run individually  👍

**Review by sarahmorrisokeefe — APPROVED** [review]

**Pull request merged**